docs/architecture/decisions/0025-caching

```diff
@@ -0,0 +1,71 @@
+
+Details:
+Originally, Cloudfront was utilized to provide caching capabilities in our application.  All incoming HTTP requests first go through a Cloudfront endpoint, which has a caching infrastructure enabled by default. Cloudfront then decides whether to pass each request to our running Django app inside cloud.gov or if it will respond to with cached data.  The big problem with this feature is Cloudfront's caching has a default timeout of 24-hours, which we cannot control.  This led to issues on our November launch;  Incidents reported include the following...
+ - Users couldn't utilize login.gov properly and had to wait a day before they would be able to login.  This was traced back to the 24-hour cache timeout.
+ - Changes made by admins would not be reflected in the app (due to the cached data not updating)
+
+To resolve these issues, we added "no cache" headers throughout our application.  Currently, every single HTTP response that comes from Django says "Cache control: no cache" in the headers, which instructs Cloudfront not to cache the associated data.  This effectively removes Cloudfront caching for us.
+
+Although we could leave our architecture as-is, we decided to investigate options for improving our use of caching (instead of just disabling it completely). 
+
+## Considered Options
+
+**Option 1:** Cache static resources using Whitenoise
+
+Caching static resources should pose little risk to our application's functionality.  Currently, every static resource from /public/... is hitting our Django application inside of Cloud.gov. We already use a Django plugin called whitenoise that can do hash-based linking to static assets so that they can be cached forever by Cloudfront. (If the content changes, then the hash changes, then it results in a different filename.)
+
+See ticket [#1371](https://github.com/cisagov/manage.get.gov/issues/1371)for more information. 
+
+**Option 2:** Leave things as-is (we had some discussion on whether or not caching staic pages will make enough of a difference to be worth the effort)
+
+
+## Decision
+
+We decided on Option 2 - leave things as-is (for now).
+
+Preliminary analysis suggest that implementing caching on static pages will result in negligible improvements to our application load time.  A quick look at Kibana logs suggests most of these resources take less than 10ms to load...
+![Kibana RTR Logs](../doc-images/caching-rtr-logs.png)
+
+If we look at average load times in Kibana, it looks like we are doing great for load times in stable (using the rtr.response_time_ms metric), staying under 200ms (in the last 4 weeks) and usually hovering around 40-80ms. Some google searching suggests that "an ideal page load time is between 0-2 seconds, but 3 seconds is also considered to be an acceptable score. Anything above 3 seconds increases the likelihood of visitors leaving your site." (Quote shamelessly copied from Sematex)
```

Comment:
**Comment:** Would you mind adding the Kibana query you used in here?

Also love the amount of detail with the numerical time metrics!

Reply:
added a hyperlink :)